### PR TITLE
fix: Make SalesforceWrapper safer

### PR DESCRIPTION
**Pull request by lidimayra**
The type of this PR is: **Fix**

### Description

Avoid hitting errors when the expected settings are not present. For ref, see:
https://artsy.slack.com/archives/C9RK0BLEP/p1687972679348039?thread_ts=1687970942.807569&cid=C9RK0BLEP

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/SalesforceWrapper.tsx [hunk 1]
```diff
@@ -57,27 +57,29 @@ export const SalesforceWrapper: React.FC<SalesforceWrapperProps> = ({
     )}/embeddedservice/5.0/esw.min.js`,
     removeOnUnmount: true,
     onReady: () => {
-      window.embedded_svc.settings.defaultMinimizedText = "Chat"
-      window.embedded_svc.settings.enabledFeatures = ["LiveAgent"]
-      window.embedded_svc.settings.entryFeature = "LiveAgent"
+      if (window.embedded_svc?.settings?.enabledFeatures) {
```
Comment (damassi) [inline src/Components/SalesforceWrapper.tsx:60]: I think if we check against settings that should enable all of the rest of the props below to be guarded safely:

```
if (window.embedded_svc?.settings) {
  ...
}
```

It doesn't look like there's anything deeper than `settings.foo`, so all good 👍

**Review by joeyAghion — DISMISSED** [review]

**Review by damassi — DISMISSED** [review]
LGTM 👍

File: src/Apps/Order/__tests__/OrderApp.jest.tsx [hunk 2]
```diff
@@ -678,7 +678,7 @@ describe("OrderApp", () => {
       mockGetENV.mockImplementation(() => ({ SALESFORCE_CHAT_ENABLED: true }))
       const props = getProps()
       const subject = getWrapper({
-        props: { ...props, location: { query: { isModal: true } } },
+        props: { ...props, match: { location: { query: { isModal: true } } } },
```
Comment (damassi) [inline src/Apps/Order/__tests__/OrderApp.jest.tsx:681]: Code in the component is pulling from `match`, so not sure how this ever passed before 🤷

Comment (lidimayra) [reply src/Apps/Order/__tests__/OrderApp.jest.tsx:681]: wow, that's interesting!! Thank you for identifying it and fixing it!! 🙇‍♀️

**Pull request merged**